### PR TITLE
chore: Miscellaneous updates mostly wrt. offline drift exploration

### PR DESCRIPTION
**Pull request by robinholzi**
In my offline exploration, I tried training a yearbook binary discriminator drift detection model. Also for the NLP datasets I added support for a preprocessing step, which is not necessary for the modyn pipeline, but useful for the offline exploration.

### CODE REVIEW

File: benchmark/huffpost_kaggle/analytics.ipynb [hunk 1]
```diff
@@ -1,110 +0,0 @@
-{
```
Comment (robinholzi) [inline benchmark/huffpost_kaggle/analytics.ipynb:1]: Moved to separate repo for offline exploration

File: modyn/supervisor/internal/triggers/drift/detector/alibi.py [hunk 2]
```diff
@@ -42,18 +46,18 @@ def init_detector(self) -> None:
...
-        embeddings_ref: pd.DataFrame | np.ndarray | torch.Tensor,
-        embeddings_cur: pd.DataFrame | np.ndarray | torch.Tensor,
+        embeddings_ref: pd.DataFrame | pd.Series | np.ndarray | torch.Tensor,
+        embeddings_cur: pd.DataFrame | pd.Series | np.ndarray | torch.Tensor,
         is_warmup: bool,
     ) -> dict[str, MetricResult]:
-        assert isinstance(embeddings_ref, (np.ndarray | torch.Tensor))
-        assert isinstance(embeddings_cur, (np.ndarray | torch.Tensor))
-        embeddings_ref = (
-            embeddings_ref.detach().cpu().numpy() if isinstance(embeddings_ref, torch.Tensor) else embeddings_ref
```
Comment (robinholzi) [inline modyn/supervisor/internal/triggers/drift/detector/alibi.py:52]: TODO: restore

File: benchmark/huffpost_kaggle/analytics.ipynb (hunk 1 above)
Comment (MaxiBoether) [reply benchmark/huffpost_kaggle/analytics.ipynb:1]: Would be cool if you could share this repo with me

File: modyn/config/schema/pipeline/trigger/drift/alibi_detect.py [hunk 3]
```diff
@@ -65,6 +69,13 @@ def validate_threshold_permutations(self) -> "AlibiDetectMmdDriftMetric":
         return self
 
 
+class AlibiDetecClassifierDriftMetric(_AlibiDetectBaseDriftMetric, AlibiDetectDeviceMixin):
```
Comment (MaxiBoether) [inline modyn/config/schema/pipeline/trigger/drift/alibi_detect.py:72]: typo: Detec

File: modyn/supervisor/internal/triggers/drift/detector/alibi.py [hunk 4]
```diff
@@ -42,18 +46,18 @@ def init_detector(self) -> None:
 
     def detect_drift(
         self,
-        embeddings_ref: pd.DataFrame | np.ndarray | torch.Tensor,
-        embeddings_cur: pd.DataFrame | np.ndarray | torch.Tensor,
+        embeddings_ref: pd.DataFrame | pd.Series | np.ndarray | torch.Tensor,
+        embeddings_cur: pd.DataFrame | pd.Series | np.ndarray | torch.Tensor,
```
Comment (MaxiBoether) [inline modyn/supervisor/internal/triggers/drift/detector/alibi.py:50]: Why can the embeddings be a series now as well?

**Review by MaxiBoether — APPROVED** [review]
Just 3 small comments, otherwise LGTM

File: modyn/supervisor/internal/triggers/drift/detector/alibi.py (hunk 4 above)
Comment (robinholzi) [reply modyn/supervisor/internal/triggers/drift/detector/alibi.py:50]: not within Modyn, but in my offline exploration (iirc after PCA) I ended up with a series for detection. I didn't see any reason not to support this directly in the detector so I added support for it.

File: benchmark/huffpost_kaggle/analytics.ipynb (hunk 1 above)
Comment (robinholzi) [reply benchmark/huffpost_kaggle/analytics.ipynb:1]: I tried to upload it to GitHub. There are many big evaluation artifact files (> 3 GB total). I tried to use LFS, but GitHub limits me to 1GB. Should I just put the code into VC and share the artifacts over nfs at the end?

**Pull request merged**